Wrappers/Python/ccpi/viewer/CILViewer2D.py

```diff
@@ -517,10 +545,30 @@ def OnROIModifiedEvent(self, interactor, event):
         bounds = pd.GetBounds()
 
         # Set the values of the ll and ur corners
-        ll = (bounds[0], bounds[2], bounds[4])
-        ur = (bounds[1], bounds[3], bounds[5])
-        vox1 = self.createVox(ll)
-        vox2 = self.createVox(ur)
+        box_voxel_min, box_voxel_max = self.GetVoxelsFromExtent(bounds)
+        box_voxel_min = list(box_voxel_min)
+        box_voxel_max = list(box_voxel_max)
+        # Get maximum extents of the image in world coords
+        data_extent = self.GetDataExtentInWorld()
+
+        voxel_min_world, voxel_max_world = self.GetVoxelsFromExtent(data_extent)
+        orientation = self.GetSliceOrientation()
+        i = [orientation, (orientation + 1) % 3, (orientation + 2) % 3]
+        if box_voxel_min[i[1]] < voxel_min_world[i[1]]:
+            box_voxel_min[i[1]] = voxel_min_world[i[1]]
+        if box_voxel_min[i[2]] < voxel_min_world[i[2]]:
+            box_voxel_min[i[2]] = voxel_min_world[i[2]]
+        if box_voxel_max[i[1]] > voxel_max_world[i[1]]:
+            box_voxel_max[i[1]] = voxel_max_world[i[1]]
+        if box_voxel_max[i[2]] > voxel_max_world[i[2]]:
+            box_voxel_max[i[2]] = voxel_max_world[i[2]]
```

Comment:
Why do you need 3 elements in `i` if you only access 2?

Reply:
You are right, I copied it from the other bit of code to be consistent. I will edit so it is only defined once.